### PR TITLE
RPC fix: switch from Redis to Postgres for indexing

### PR DESCRIPTION
**Pull request by ngundotra**
We need generalized asset indexer to be built in a way that makes NFT RPC calls easy to implement on top of the pre existing indexes as according to sRFC #16.

### CODE REVIEW

File: tests/nftRpc.ts [hunk 1]
```diff
@@ -1,39 +1,71 @@
...
+  async fetchNFTsForAuthority(authority: PublicKey): Promise<AssetGroup[]> {
+    // return await this.gIndexer.agRepo
+    //   .search()
+    //   .where("authority")
+    //   .equals(authority.toBase58())
+    //   .and("pubkeys")
+    //   .contains(authority.toBase58())
+    //   .returnAll();
+
+    // TODO: fix
```
Comment (buffalojoec) [inline tests/nftRpc.ts]: Next PR for this? Lol

**Review by buffalojoec — COMMENTED** [review]
Lgtm mate, but do we want to include the Redis stuff still? I'm thinking we should cut it out and drop the deps from `package.json`, and then you can decide if the indexer should still have the `Pg` suffix

File: package.json [hunk 2]
```diff
@@ -10,10 +10,11 @@
         "@types/mocha": "^9.0.0",
         "chai": "^4.3.4",
         "mocha": "^9.0.3",
+        "pg": "^8.11.0",
         "prettier": "^2.6.2",
         "redis": "^4.6.7",
         "redis-om": "^0.3.6",
```
Comment (buffalojoec) [inline package.json]: Maybe drop `redis` and `redis-om` from the deps if we aren't including the indexer anymore?

Comment (ngundotra) [reply package.json]: Updated 🖖

File: tests/nftRpc.ts (hunk 1 above)
Comment (ngundotra) [reply tests/nftRpc.ts]: fixed 😂

**Comment by ngundotra** [comment]
@buffalojoec I think the missing stuff is putting `emit_collection!` and `emit_metadata!` macros in the `nft_events` library.

**Pull request merged**